### PR TITLE
Extend ledger init with vacation requests/approvals

### PR DESCRIPTION
**Pull request by shayne-fletcher**

### CODE REVIEW

File: admin-cli/src/index.ts [hunk 1]
```diff
@@ -5,10 +5,17 @@ import * as davl3 from '@daml2ts/davl-v3/lib/edb5e54da44bc80782890de3fc58edb5cc2
...
+  fromDate: string;
+  toDate: string;
+  approved: boolean;
+}
+
 type EmployeeInfo = {
   boss: string;
   vacationDays: number;
   acceptProposal: boolean;
+  vacationRequests: Vacation[];
```
Comment (hurryabit) [inline admin-cli/src/index.ts]: Let's make this optional. This makes the config file easier to use.
```suggestion
  vacationRequests?: Vacation[];
```

File: admin-cli/src/index.ts [hunk 2]
```diff
@@ -61,7 +70,41 @@ namespace v3 {
         console.log(`Accepted EmployeeProposal for ${employee}.`);
       }
     }
+
+    // Create/approve employee vacation requests.
+    for (const employee in config.employees) {
+      const employeeInfo = config.employees[employee];
+      for (const vacationRequest of employeeInfo.vacationRequests) {
```
Comment (hurryabit) [inline admin-cli/src/index.ts]: Assuming the make the `vacationRequests` field optional:
```suggestion
      for (const vacationRequest of employeeInfo.vacationRequests ?? []) {
```

File: admin-cli/src/index.ts [hunk 3]
```diff
@@ -61,7 +70,41 @@ namespace v3 {
...
+        const employeeLedger = connect(config, employee);
+        const employeeRoleContract = await employeeLedger.lookupByKey(davl3.EmployeeRole, employee);
+        if (employeeRoleContract) {
+          const pendingRequest =
```
Comment (hurryabit) [inline admin-cli/src/index.ts]:
```suggestion
          const [vacationRequestContractId] =
```

File: admin-cli/src/index.ts [hunk 4]
```diff
@@ -61,7 +70,41 @@ namespace v3 {
...
+            await employeeLedger.exercise(
+              davl3.EmployeeRole.EmployeeRole_RequestVacation,
+              employeeRoleContract.contractId,
+              { fromDate: vacationRequest.fromDate, toDate: vacationRequest.toDate },
+          );
+          console.log(
+            'Created VacationRequest [' +
+              vacationRequest.fromDate + ', ' + vacationRequest.toDate + `] for ${employee}`);
+          if (vacationRequest.approved){
+            const vacationRequestContractId = pendingRequest[0];
```
Comment (hurryabit) [inline admin-cli/src/index.ts]: Assuming we do what I suggested above.
```suggestion
```

File: admin-cli/src/index.ts [hunk 5]
```diff
@@ -61,7 +70,41 @@ namespace v3 {
...
+            const boss = employeeInfo.boss
+            const bossLedger = connect(config, boss);
+            await bossLedger.exercise(
+              davl3.VacationRequest.VacationRequest_Accept,
+              vacationRequestContractId,
+              {},
```
Comment (hurryabit) [inline admin-cli/src/index.ts]: What do you think about changing `@daml/ledger` such that the third argument is optional when the choice argument type is `{}`? Would that feel better?

File: admin-cli/src/index.ts [hunk 6]
```diff
@@ -61,7 +70,41 @@ namespace v3 {
...
+            );
+            console.log(
+              `${boss} approved VacationRequest [` +
```
Comment (hurryabit) [inline admin-cli/src/index.ts]: Either we do string interpolation or we don't. 😉
```suggestion
              `${boss} approved VacationRequest \
```

File: admin-cli/src/index.ts [hunk 7]
```diff
@@ -61,7 +70,41 @@ namespace v3 {
...
+                vacationRequest.fromDate + ', ' + vacationRequest.toDate + `] for ${employee}`);
```
Comment (hurryabit) [inline admin-cli/src/index.ts]:
```suggestion
                [${vacationRequest.fromDate}, ${vacationRequest.toDate}] for ${employee}`);
```

File: test-setup.json [hunk 8]
```diff
@@ -8,22 +8,29 @@
     "Martin": {
       "boss": "Shaul",
       "vacationDays": 25,
-      "acceptProposal": false
+      "acceptProposal": false,
+      "vacationRequests": []
```
Comment (hurryabit) [inline test-setup.json]: If we follow my suggestion to make this field optional.
```suggestion
```

File: admin-cli/src/index.ts (hunk 3 above)
Comment (shayne-fletcher) [reply admin-cli/src/index.ts]: But pending request is of form `[_, _]`. If I write `[a, _]` I get an unused variable warning on the `_`.

File: admin-cli/src/index.ts (hunk 5 above)
Comment (shayne-fletcher) [reply admin-cli/src/index.ts]: Explicit is better than implicit. I see require the argument even when it's unit.

**Comment by shayne-fletcher** [comment]
I'll take the optional. Wait on the next iteration on this PR. Have questions about `pendingRequest` though.

File: admin-cli/src/index.ts (hunk 3 above)
Comment (hurryabit) [reply admin-cli/src/index.ts]:
```typescript
const [x] = [1, 2];
```
is perfectly valid TypeScript. The `2` simply doesn't get bound to a name. In other words, simply skip the `_`.

Comment (shayne-fletcher) [reply admin-cli/src/index.ts]: Ok!

**Pull request merged**